client/go/outline/platerrors/platform_error.go

```diff
@@ -34,6 +35,17 @@ type PlatformError struct {
 
 var _ error = PlatformError{}
 
+// NewPlatformError creates a new [PlatformError] from the error code and message.
+//
+// This function is primarily intended for use by Java or Swift code.
+// For Go code, it is recommended to construct the [PlatformError] struct directly.
+func NewPlatformError(code ErrorCode, message string) *PlatformError {
```

Comment:
No need to change in this PR, but I find the name `PlatformError` confusing. We use that name for the platform-specific errors going back to Typescript too.
Perhaps it should be `GoError`, since it's the type we use for the Go code.

Reply:
Actually I intentionally make TypeScript's `PlatformError` identical to Go's `PlatformError`, so devs would know these errors are the same.